### ISSUE TITLE
Accessibility seems to be broken for iframes embedded inside webview

### ISSUE THREAD

**Issue by sarthak-saxena** [electron/electron#24589]
* **Electron Version(s)**
  * `1.3.1`, `1.8.4`, `8.4.0`, `9.1.0` & `10.0.0-beta.12`
* **Operating System:**
  * MacOS
* **Last Known Working Electron version:**
  * NA

### Expected Behavior
<img width="551" alt="Screen Shot 2020-07-16 at 4 21 27 PM" src="https://user-images.githubusercontent.com/9450414/87662802-69068e00-c780-11ea-8aea-426c189b3a9a.png">
I have an app which has 3 levels, level1 incudes a webview with page source level2.html; inside level2 an `<iframe>` element is embedded with source set to level3.html. Each page has button & input field.

The expected behavior would be while tabbing the screen readers should be able to focus & read through all three buttons & input fields.

### Actual Behavior
Screen readers are reaching till the buttons & input fields of level 2 & ignoring level3. But if I relace `<webview>` with `<iframe>` everything seems to work fine.
[Please check out the video for more details on the issue](https://www.loom.com/share/318395aa07f54592bd047f6afa4b300a)

**Check the below image showing the accessibility tree not being generated properly incase of webview**:
<img width="977" alt="Screen Shot 2020-07-16 at 4 36 20 PM" src="https://user-images.githubusercontent.com/9450414/87664530-0498fe00-c783-11ea-8b27-cfff4ad1eedd.png">

### To Reproduce
- Clone this repository
   $ git clone https://github.com/microsoft/electron-webview-accessibility-issue
   $ cd electron-quick-start
   $ npm install
   $ npm start || electron .

- Turn on the screen reader
- Tab across button & input fields
- You will notice that the screen reader focus is not moving to level3 hence is not able to read the corresponding elements

### Additional Information
This issue has been reported earlier & is a possible duplicate of https://github.com/electron/electron/issues/12478 but it contains more information on reproducing the bug  for electron version `8.4.0`

**Comment by Kilian** [comment]
Apparently this is an upstream Chromium bug because of out of process iframes and we will probably have to wait for them to fix it.

**Comment by petewu100** [comment]
Kilian, is there a bug report on Chromium tracking this issue?  Can you add a link to it here?  Thank you.

**Comment by BuckeyeJL** [comment]
This is the Chromium bug:
https://bugs.chromium.org/p/chromium/issues/detail?id=1130710

**Comment by petewu100** [comment]
Thank you @BuckeyeJL.  The Chromium bug had a few updates in the past week:

10/13/2020 https://chromium.googlesource.com/chromium/src.git/+/e1ec154666b2fdeab0cbee3ec31f1d3fbea43770 Focus WebContents prior to sending focus to renderer for accessibility
10/14/2010 https://chromium.googlesource.com/chromium/src.git/+/e9be7d1826ea5952232615255677fa6406a67053 Revert of the previous commit
10/21/2020 https://chromium.googlesource.com/chromium/src/+/50d1c1ea5382add579d292827931e1f0aec499b7 Ensure the root BrowserAccessibility AXTreeData is updated for focus

What are the next steps on integrating this with Electron and testing whether it fixes the problem?  Thank you!

**Comment by ckerr** [comment]
@petewu100 the first thing to do is to confirm that this is working now in Chromium, e.g. to try the nested [index..index3].html files from @sarthak-saxena's https://github.com/microsoft/electron-webview-accessibility-issue repro.

If this _is_ working in Chromium due to those upstream fixes, I can try cherry-picking them back into the supported versions of Electron

**Comment by ckerr** [comment]
Hey, I just wanted to do a quick checkin on this ticket to say that I'm going to be offline for the rest of the week but this ticket is at the top of my TODO for Monday.

**Comment by ckerr** [comment]
The three commits [mentioned above](https://github.com/electron/electron/issues/24589#issuecomment-713901873) appear to not be sufficient to fix the issue. All three are included in v13.0.0-nightly.20201130, and testing https://github.com/microsoft/electron-webview-accessibility-issue with that build still yields the `generic` name rather than `Iframe`.

So, I'm going to try and figure out why we're getting a ax::mojom::blink::Role::kGenericContainer role there rather than ax::mojom::blink::Role::kIframe

**Comment by petewu100** [comment]
Any update on this? Are there still issues on Electron that needs to be fixed (even after picking up the Chromium fixes) to make this work?  Thank you.

**Comment by ckerr** [comment]
@petewu100 yes! this is still a work in progress :slightly_smiling_face: Unfortunately those upstream Chromium fixes do not fix this issue in Electron and I am investigating why.

**Comment by ckerr** [comment]
This was not working in v13.0.0-nightly.20201130, which included the upstream fixes, but is working in v13.0.0-nightly.20210112.

I will backport those three fixes and also bisect to see what other items are needed that changed the behavior between 01130 and 10112.

**Comment by ckerr** [comment]
For visibility, here's an update on the tracking:

This is broken in v13.0.0-nightly.20201214 and fixed in v13.0.0-nightly.20201215. There are only a few commits between those two tags:

```sh
git log --pretty=oneline v13.0.0-nightly.20201214..v13.0.0-nightly.20201215
ef49fea5eb4817db684ee2c590cb9b7e003f62f5 (tag: v13.0.0-nightly.20201215) Bump v13.0.0-nightly.20201215
9b02d94e9767d67a7b176bffc009e70fe5767b7a fix(asar): readdir(withFileTypes) fails on deep directory (#26865)
b73b343e29e86e5361a94cdddcc0f1463b247c2a docs: update quick start description after fiddle integration (#26943)
4588a411610ee4095ab2a47e086f23fa4730e50e (HEAD) chore: bump chromium to 096e5313aaf19dfa0c4710145c34d (master) (#26535)
```

And as you'd expect from that list, 4588a411610ee4095ab2a47e086f23fa4730e50e is the commit that fixes it. I've built that commit to confirm the fix. In that commit, the Chromium DEPS is bumped from [3a75ada](https://chromium-review.googlesource.com/q/3a75ada69d1ac06d6903a2c981ab90a8162f1ba0) to [89292a4](https://chromium-review.googlesource.com/q/89292a4ae29096e5313aaf19dfa0c4710145c34d) which covers commits in [2020-11-16...2020-12-07].

Of the commits in this range, these touch ax_object.cc and so one or more of these is the likely fix.

* https://chromium-review.googlesource.com/c/chromium/src/+/2575445
* https://chromium-review.googlesource.com/c/chromium/src/+/2568365
* https://chromium-review.googlesource.com/c/chromium/src/+/2566054
* https://chromium-review.googlesource.com/c/chromium/src/+/2563861
* https://chromium-review.googlesource.com/c/chromium/src/+/2485519
* https://chromium-review.googlesource.com/c/chromium/src/+/2555957
* https://chromium-review.googlesource.com/c/chromium/src/+/2553695
* https://chromium-review.googlesource.com/c/chromium/src/+/2548465
* https://chromium-review.googlesource.com/c/chromium/src/+/2538117
